presto-spi/src/main/java/io/prestosql/spi/security/ViewExpression.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.spi.security;
+
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class ViewExpression
+{
+    private final String identity;
```

Comment:
The identity attached to the filter is the identity that should be used to evaluate any tables referenced by the filter. If you ask: “give me a filter for table T for Bob”, you might get “here’s the filter, and any tables referenced in sub queries in that filter should be resolved as user Admin”. This can be useful to constrain permissions based on data contained in lookup tables only accessible to users with more privileges.

It’s no different that how views work.

Reply:
Makes sense. Thanks